src/mon/OSDMonitor.h

```diff
@@ -303,6 +312,151 @@ class OSDMonitor : public PaxosService {
+  double cache_inc_ratio = 0;       ///< Cache ratio dedicated to inc
+  double cache_full_ratio = 0;      ///< Cache ratio dedicated to full
+  uint64_t mon_memory_base = 0;     ///< Mon base memory for cache autotuning
+  double mon_memory_fragmentation = 0; ///< Expected memory fragmentation
+  uint64_t mon_memory_target = 0;   ///< Mon target memory for cache autotuning
+  uint64_t mon_memory_min = 0;      ///< Min memory to cache osdmaps
+  bool mon_memory_autotune = false; ///< Cache auto tune setting
+  uint64_t max_inc_cache = 0;
+  uint64_t max_full_cache = 0;
+  int register_cache_with_pcm();
+  int _set_cache_sizes();
+  int _set_cache_ratios();
+  void _set_new_cache_sizes();
+  uint64_t calculate_used_cache_bytes(bool cache_type_full);
+  void _set_cache_autotuning();
+  int _update_mon_cache_settings();
+
+  struct OSDMemCache : public PriorityCache::PriCache {
+    OSDMonitor *osdmon;
+    int64_t cache_bytes[PriorityCache::Priority::LAST+1] = {0};
+    int64_t committed_bytes = 0;
+    double cache_ratio = 0;
+    uint64_t used_bytes = 0;
+
+    OSDMemCache(OSDMonitor *m) : osdmon(m) {};
+
+    virtual uint64_t _get_used_bytes() const = 0;
+    virtual void _update_used_bytes(size_t size, bool recalc) = 0;
+
+    virtual int64_t request_cache_bytes(
+        PriorityCache::Priority pri, uint64_t total_cache) const {
+      int64_t assigned = get_cache_bytes(pri);
+
+      switch (pri) {
+      // All cache items are currently set to have PRI1 priority
+      case PriorityCache::Priority::PRI1:
+        {
+          int64_t request = _get_used_bytes();
+          return (request > assigned) ? request - assigned : 0;
+        }
+      default:
+        break;
+      }
+      return -EOPNOTSUPP;
+    }
+
+    virtual int64_t get_cache_bytes(PriorityCache::Priority pri) const {
+        return cache_bytes[pri];
+    }
+
+    virtual int64_t get_cache_bytes() const {
+      int64_t total = 0;
+
+      for (int i = 0; i < PriorityCache::Priority::LAST + 1; i++) {
+        PriorityCache::Priority pri = static_cast<PriorityCache::Priority>(i);
+        total += get_cache_bytes(pri);
+      }
+      return total;
+    }
+
+      virtual void set_cache_bytes(PriorityCache::Priority pri, int64_t bytes) {
+        cache_bytes[pri] = bytes;
+      }
+      virtual void add_cache_bytes(PriorityCache::Priority pri, int64_t bytes) {
+        cache_bytes[pri] += bytes;
+      }
+      virtual int64_t commit_cache_size(uint64_t total_cache) {
+        committed_bytes = PriorityCache::get_chunk(
+            get_cache_bytes(), total_cache);
+        return committed_bytes;
+      }
+      virtual int64_t get_committed_size() const {
+        return committed_bytes;
+      }
+      virtual double get_cache_ratio() const {
+        return cache_ratio;
+      }
+      virtual void set_cache_ratio(double ratio) {
+        cache_ratio = ratio;
+      }
+      virtual string get_cache_name() const = 0;
```

Comment:
is this method used anywhere?

Reply:
It's not used anywhere currently. I suppose it could be used for logging/diagnosis in future.